File: Programmers/가운데 글자 가져오기.js
// https://programmers.co.kr/learn/courses/30/lessons/12903?language=javascript

function solution(s) {
  return s.slice(Math.ceil(s.length / 2) - 1, Math.floor(s.length / 2 + 1));
}

const s = "abcde"; // c    // 2,3
const a = "qwer"; // we   // 1,3

console.log(solution(s));
console.log(solution(a));

File: Programmers/가운데 글자 가져오기.swift
//https://programmers.co.kr/learn/courses/30/lessons/12903?language=swift

import Foundation

func solution(_ s:String) -> String {
    var answer: String = ""
    let midLeft: Int = Int(floor(Double(s.count - 1) / 2))
    let midRight: Int = Int(ceil(Double(s.count - 1) / 2))
    
    for i in midLeft...midRight {
        answer += String(s[s.index(s.startIndex, offsetBy: i)])
    }
    
    return answer
}

let sa: String = "abcde"
//"c"

let sb: String = "qwer"
//"we"

print(solution(sa))
print(solution(sb))
